### PR TITLE
Fix some errors reading C-DNS RFC files.

### PR DESCRIPTION
**Pull request by banburybill**

### CODE REVIEW

File: lib/cdns.cpp [hunk 1]
```diff
@@ -2966,7 +2966,7 @@ uint8_t* cdnsStorageParameter::parse_map_item(uint8_t* in, uint8_t const* in_max
     }
 
     if (in == NULL) {
-        fprintf(stderr, "Cannot parse cdnsCollectionParameters %d, err=%d\n", (int)val, *err);
+        fprintf(stderr, "Cannot parse cdnsStorageParameters %d, err=%d\n", (int)val, *err);
     }
```
Comment (huitema) [inline lib/cdns.cpp:2970]: My bad... Thanks for fixing that.

File: lib/cdns.h [hunk 2]
```diff
@@ -398,10 +398,10 @@ class cdnsCollectionParameters
...
     int64_t snaplen;
     bool promisc;
-    std::vector<cbor_bytes> interfaces;
+    std::vector<cbor_text> interfaces;
     std::vector<cbor_bytes> server_addresses;
-    std::vector<cbor_bytes> vlan_id;
-    cbor_bytes filter;
+    std::vector<int> vlan_id;
+    cbor_text filter;
     int64_t query_options;
```
Comment (huitema) [inline lib/cdns.h:405]: Yes, that's what the spec says. I missed that!

**Review by huitema — APPROVED** [review]
Thanks for the fixes, @banburybill. Would you be able to share a small CDNS file that actually uses the interfaces and vlan-id parameters? That would help write a non-regression test.

**Comment by banburybill** [comment]
Hi Christian,

On Thursday, 29 October 2020 00:25:06 GMT Christian Huitema wrote:
> Thanks for the fixes, @banburybill. Would you be able to share a small CDNS
> file that actually uses the interfaces and vlan-id parameters? That would
> help write a non-regression test.

https://github.com/dns-stats/compactor/blob/master/test-scripts/gold.cbor.xz
has vlan_id and filter entries.

I'll see if I can get a capture will those and interface entries as well 
today.
-- 
Jim Hague
jim@sinodun.com                      Never trust a computer you can't lift.
Sinodun Internet Technologies Ltd.   +44 7941 697732

**Comment by banburybill** [comment]
On Thursday, 29 October 2020 09:16:56 GMT Jim Hague wrote:
> https://github.com/dns-stats/compactor/blob/master/test-scripts/gold.cbor.xz
> has vlan_id and filter entries.
> 
> I'll see if I can get a capture will those and interface entries as well
> today.

Here's a modified version of the above file with the parameter structures 
filled out. Some of these currently aren't generated by compactor...
-- 
Jim Hague
jim@sinodun.com                      Never trust a computer you can't lift.
Sinodun Internet Technologies Ltd.   +44 7941 697732

**Pull request merged**